lib/rules/base.js

```diff
@@ -520,17 +511,35 @@ module.exports = class Base {
   }
 
   log(result) {
+    let locProps = {};
     let defaults = {
       rule: this.ruleName,
       severity: this.severity,
     };
+
     if (this._filePath) {
       defaults.filePath = this._filePath;
     }
+
     if (this[MODULE_NAME]) {
       defaults.moduleId = this[MODULE_NAME];
     }
-    let reportedResult = Object.assign({}, defaults, result);
+
+    // perform the node prop expansion only if those properties don't exist in result
+    if (result.node) {
+      let node = result.node;
+      let nodeProps = {
+        line: node.loc && node.loc.start.line,
+        column: node.loc && node.loc.start.column,
+        source: this.sourceForNode(node),
+      };
+
+      delete result.node;
+
+      locProps = Object.assign({}, nodeProps, result);
+    }
+
+    let reportedResult = Object.assign({}, defaults, locProps, result);
```

Comment:
I don't _think_ we need two assign's here 🤔

Reply:
That's true, we can just do the single assign on 542.

Reply:
Done.